spec/mysql2/client_spec.rb

```diff
@@ -963,6 +963,20 @@ def run_gc
     end.not_to raise_error
   end
 
+  context 'with query-level timeout' do
+    it "time outs on long queries" do
+      expect do
+        @client.query("select sleep(2)", read_timeout: 1)
+      end.to raise_error(Mysql2::Error::TimeoutError)
+    end
+
+    it "does not time out on queries under timeout" do
+      expect do
+        client.query("select sleep(1)", read_timeout: 2)
```

Comment:
I think you're after the instance variable here :smile:

Reply:
Thanks @jacobbednarz!